### PR TITLE
677 - Update meta details in readme

### LINKED ISSUE #677: Create readme for nodehandler / meta changes
Need to add readme for recent changes to nodehandler / meta / dimensions

### PR DESCRIPTION
**Pull request by tomdye**
**Type:** bug / feature

The following has been addressed in the PR:

* [x] There is a related issue

**Description:**

Adds details for the `meta` and `nodehandler` changes to the readme file.

Resolves #677

### CODE REVIEW

**Comment by agubler** [comment]
There is an issue I believe, can we update the description.

File: README.md [hunk 1]
```diff
@@ -1028,21 +1027,18 @@ class MyWidget extends WidgetBase<WidgetProperties> {
...
     // ...
 }
 ```
 
-Meta classes are provided with a few hooks into the widget, passed to the constructor:
+Extending the base class found in `meta/Base` will provide the following functionality:
 
-* `nodes` - A map of `key` strings to DOM elements. Only `v` nodes rendered with `key` properties are stored.
-* `requireNode` - A method that accepts a `key` string to inform the widget it needs a rendered DOM element corresponding to that key. If one is available, it will be returned immediately. If not, the widget will be re-rendered and if the node does not exist on the next render, an **error** will be thrown.
+* `has` - A method that accepts a `key` and returns a `boolean` to denote if that node exists in the rendered DOM.
```
Comment (agubler) [inline README.md]: `if that` read a bit oddly => `if the node for the request key exist in the DOM`

File: README.md [hunk 2]
```diff
@@ -1028,21 +1027,18 @@ class MyWidget extends WidgetBase<WidgetProperties> {
...
+* `getNode` - A method that accepts a `key` string to inform the widget it needs a rendered DOM element corresponding to that key. If one is available, it will be returned immediately. If not, a callback is created which will invalidate the widget when the node becomes available. This uses the underlying `nodeHandler` event system.
 * `invalidate` - A method that will invalidate the widget.
-
-Extending the base class found in `meta/Base` will automatically add these hooks to the class instance as well as providing a `has` method:
-
-* `has(key: string)` - A method that returns `true` if the DOM element with the passed `key` exists in the rendered DOM.
+* `afterRender` - A function that is called after the render is complete. This can be used to clear down any maps etc... that a meta may have created. This is used, for instance, in the `WebAnimation` meta to clear down unused animations.
```
Comment (agubler) [inline README.md]: Perhaps something like this - `This provides a hook into the widget `afterRender` lifecycle that can be used to clear up any resources that the meta has created.`

File: README.md (hunk 1 above)
Comment (tomdye) [reply README.md]: sure

**Comment by tomdye** [comment]
@agubler comments addressed

File: README.md (hunk 2 above)
Comment (tomdye) [reply README.md]: done

File: README.md [hunk 3]
```diff
@@ -1028,21 +1027,18 @@ class MyWidget extends WidgetBase<WidgetProperties> {
...
+* `has` - A method that accepts a `key` and returns a `boolean` to denote if the corresponding node exists in the rendered DOM.
+* `getNode` - A method that accepts a `key` string to inform the widget it needs a rendered DOM element corresponding to that key. If one is available, it will be returned immediately. If not, a callback is created which will invalidate the widget when the node becomes available. This uses the underlying `nodeHandler` event system.
 * `invalidate` - A method that will invalidate the widget.
-
-Extending the base class found in `meta/Base` will automatically add these hooks to the class instance as well as providing a `has` method:
-
-* `has(key: string)` - A method that returns `true` if the DOM element with the passed `key` exists in the rendered DOM.
+* `afterRender` - This provides a hook into the widget afterRender lifecycle that can be used to clear up any resources that the meta has created. This is used, for instance, in the `WebAnimation` meta to clear down unused animations.
```
Comment (agubler) [inline README.md]: might want to use ` for the afterRender reference but otherwise :+1:

**Pull request merged**